tensorflow/java/src/main/java/org/tensorflow/Input.java

```diff
@@ -0,0 +1,61 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+package org.tensorflow;
+
+/**
+ * A tensor that can be used as an operand for operation wrappers.
+ *
+ * <p>The {@code Input} interface is at the base of the Java Ops API by allowing any object that
+ * wraps an input tensor to be passed as an operand to an operation.
+ *
+ * <pre>{@code
+ * // Output implements Input.
+ * Output image = ops.image().decodeJpeg(...).image();
```

Comment:
In this case, it won't be true, since the DecodeJpeg operation cannot be cast to Output directly. I will change the type of the "image" variable for "Input" and add an additional example where an Output must be selected from multiple-output operation.